### PR TITLE
feat: [SEI-3135]: Export SEICustomMicroFrontendProps for external use

### PR DESCRIPTION
**Pull request by sharath-levelops**
### Summary

Export SEICustomMicroFrontendProps for external use

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by pankajsingh-harness — DISMISSED** [review]

File: src/microfrontends/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@harness/microfrontends",
-  "version": "2.9.0",
+  "version": "2.9.1",
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/package.json:3]: why not `2.10.0`? it's a new feature addition.

File: src/modules/75-sei/RouteDestinations.tsx [hunk 2]
```diff
@@ -86,52 +84,26 @@ export default function SEIRoutes(): React.ReactElement {
...
-          HomePageTemplate
-        }}
-        cdServices={{
-          useGetLicensesAndSummary
-        }}
-        customRoutes={routes}
-        customUtils={{ NameSchema }}
-      />
-    </RouteWithLayout>
+    <>
```
Comment (abhinavrastogi-harness) [inline src/modules/75-sei/RouteDestinations.tsx:88]: react fragment `<>` is not needed

File: src/modules/75-sei/RouteDestinations.tsx [hunk 3]
```diff
@@ -86,52 +84,26 @@ export default function SEIRoutes(): React.ReactElement {
...
+      <RouteWithLayout layout={MinimalLayout} path={routes.toSEI({ ...accountPathProps })} sidebarProps={{}}>
```
Comment (abhinavrastogi-harness) [inline src/modules/75-sei/RouteDestinations.tsx:89]: you can remove `sidebarProps` entirely

File: src/modules/75-sei/RouteDestinations.tsx (hunk 2 above)
Comment (sharath-levelops) [reply src/modules/75-sei/RouteDestinations.tsx:88]: I tried removing the fragment. Without React fragment it is not working.

File: src/modules/75-sei/RouteDestinations.tsx (hunk 3 above)
Comment (sharath-levelops) [reply src/modules/75-sei/RouteDestinations.tsx]: oh. it was added for testing. I removed it.

File: src/microfrontends/package.json (hunk 1 above)
Comment (sharath-levelops) [reply src/microfrontends/package.json]: Updated

**Review by abhinavrastogi-harness — DISMISSED** [review]

File: src/microfrontends/index.ts [hunk 4]
```diff
@@ -52,6 +52,7 @@ import type { IDPCustomMicroFrontendProps } from '@idp/interfaces/IDPCustomMicro
 import type { GovernanceCustomMicroFrontendProps } from '@governance/GovernanceCustomMicroFrontendProps.types'
 import type { ETCustomMicroFrontendProps } from '@cet/ErrorTracking.types'
 import type { SRMCustomMicroFrontendProps } from '@cv/interface/SRMCustomMicroFrontendProps.types'
+import { SEICustomMicroFrontendProps } from '@sei/SEICustomMicroFrontendProps.types'
```
Comment (mayankVermaHarness) [inline src/microfrontends/index.ts:55]: Is this supposed to be a type import?
```
import **type** { SEICustomMicroFrontendProps } from '@sei/SEICustomMicroFrontendProps.types'
```

File: src/modules/20-rbac/RouteDestinations.tsx [hunk 5]
```diff
@@ -343,115 +344,130 @@ export const AccessControlRouteDestinations: React.FC<{
...
-      path={[routes.toResourceGroupDetails({ ...projectPathProps, ...moduleParams, ...resourceGroupPathProps })]}
-      pageName={PAGE_NAME.ResourceGroupDetails}
-    >
-      <ResourceGroupDetails />
-    </RouteWithLayout>
-  </>
-)
+  layout?: React.ComponentType
+}> = ({ moduleParams, licenseRedirectData, sidebarProps, layout }) => {
+  const Layout = layout || MinimalLayout
```
Comment (mayankVermaHarness) [inline src/modules/20-rbac/RouteDestinations.tsx:349]:
```suggestion
}> = ({ moduleParams, licenseRedirectData, sidebarProps, layout = MinimalLayout }) => {
```

**Review by mayankVermaHarness — APPROVED** [review]

**Review by bhavya-sinha — DISMISSED** [review]

File: src/microfrontends/index.ts (hunk 4 above)
Comment (sharath-levelops) [reply src/microfrontends/index.ts]: Updated

**Review by harshilbits — APPROVED** [review]

**Review by ajay-levelops — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Review by knagurski — APPROVED** [review]
👍🏻

**Pull request merged**